sentry_sdk/integrations/spark/spark_worker.py

```diff
@@ -0,0 +1,69 @@
+from __future__ import absolute_import
+
+import sys
+
+from sentry_sdk import configure_scope
+from sentry_sdk.hub import Hub
+from sentry_sdk.utils import (
+    capture_internal_exceptions,
+    exc_info_from_error,
+    single_exception_from_error_tuple,
+    walk_exception_chain,
+    event_hint_with_exc_info,
+)
+
+
+def _capture_exception(exc_info, hub):
+    from pyspark.taskcontext import TaskContext
+
+    client = hub.client
+
+    client_options = client.options
+    mechanism = {"type": "spark", "handled": False}
+
+    exc_info = exc_info_from_error(exc_info)
+
+    exc_type, exc_value, tb = exc_info
+    rv = []
+
+    for exc_type, exc_value, tb in walk_exception_chain(exc_info):
```

Comment:
Why do you explicitly build the error chain here?

Reply:
Sorry should have given you some more context here. 

When a PySpark worker throws an exception, it doesn't raise any exceptions

https://github.com/apache/spark/blob/master/python/pyspark/worker.py#L599-L617

Instead it writes to the JVM, where there is Scala code to handle the Python error (log/start new worker/kill application etc.) 

https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/api/python/PythonRunner.scala#L618-L619

So the exception handling path here is that the worker will write to the stream and then call `sys.exit(-1)`.

The reason I am explicitly building the error chain is that I want to remove the `SystemExit`, `EOFError`, and `ConnectionResetError` exception types. `SystemExit` will always exist as `sys.exit(-1)` is being called on worker shut down (so provide no useful information to the user), while `EOFError` and `ConnectionResetError` are noisy exceptions that the workers raise frequently (and also are not useful as a result).

Reply:
For my understanding. Are those errors part of the same exception chain that contains legitimate errors? We filter out error types in other integrations as well but never had to do this

Reply:
Yes, these errors are part of the same exception chain that contains legitimate errors.